Source/Android/jni/MainAndroid.cpp

```diff
@@ -622,6 +623,18 @@ JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_ReloadConfig
   SConfig::GetInstance().LoadSettings();
 }
 
+JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_SetOverlaySIDevice(
+    JNIEnv* env, jobject obj, jint gcController)
+{
+  IniFile ini;
+  ini.Load(File::GetUserPath(F_DOLPHINCONFIG_IDX));  // load first to not kill unknown stuff
+  IniFile::Section* core = ini.GetOrCreateSection("Core");
+
+  core->Set("SIDevice0", gcController);
```

Comment:
This doesn't actually do anything, since you're not saving the IniFile afterwards.

Reply:
Fixed. Previously, I made the INI changes by calling `NativeLibrary.SetConfig` before `NativeLibrary.SetOverlaySIDevice`, making this dead code. Now I save the INI in `NativeLibrary.SetOverlaySIDevice` and removed the unnecessary `NativeLibrary.SetConfig` calls.